### PR TITLE
[1.14] Upgrade test for Eventing Kafka channel

### PR DESCRIPTION
**Pull request by cardil**
This PR uses the Eventing `KafkaChannel` upgrade tests that were back ported in https://github.com/openshift-knative/eventing-kafka/pull/132 in Serverless operator's upgrade test suite.

### CODE REVIEW

**Comment by mgencur** [comment]
This looks good! One of the github checks is failing. But, I figured we don't have CI for release-1.14 branch. I'm going to set this up now.

**Comment by cardil** [comment]
Thanks @mgencur for CI setup. Tests are failing because of https://github.com/knative/eventing/issues/5176

I'll add a walkaround.

File: .github/workflows/validate.yaml [hunk 1]
```diff
@@ -41,7 +41,16 @@ jobs:
...
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+            echo '::debug::Running `git status`'
+            git -c color.status=always status
+            echo '::debug::Running `git diff`'
+            git -c color.ui=always diff
+            echo '::error::Not all generated files are commited. Run `make generated-files` and commit files.'
+            echo '::warning::`make generated-files` needs to be run on GOPATH due to https://github.com/knative/pkg/issues/1287'
+            exit 33
+          fi
```
Comment (cardil) [inline .github/workflows/validate.yaml:53]: This produce the following output: https://github.com/openshift-knative/serverless-operator/pull/875/checks?check_run_id=2286742732#step:6:74

File: test/upgrade/upgrade_test.go [hunk 2]
```diff
@@ -16,20 +16,24 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package upgrade
+package upgrade_test
```
Comment (mgencur) [inline test/upgrade/upgrade_test.go:19]: Why is this package changed? It's already under `test/upgrade` so isn't it redundant?

**Review by mgencur — COMMENTED** [review]
Looks good. Just one question about the package change.

File: test/upgrade/upgrade_test.go (hunk 2 above)
Comment (cardil) [reply test/upgrade/upgrade_test.go:19]: It's a good practice in Go to write tests in the `_test` package. That disallow to test private functions, forcing all tests to use only public interfaces, and making code more testable in the end.

There's also a golangci-lint linter that enforce that https://github.com/maratori/testpackage#motivation

Comment (mgencur) [reply test/upgrade/upgrade_test.go:19]: OK. Thanks

**Pull request merged**